### PR TITLE
Port Lint to 1.17

### PR DESCRIPTION
**Pull request by sylv256**

### CODE REVIEW

File: src/main/java/me/hydos/lint/client/entity/model/TinyPotatoEntityModel.java [hunk 1]
```diff
@@ -21,21 +21,27 @@
...
 import net.minecraft.client.render.VertexConsumer;
 import net.minecraft.client.render.entity.model.CompositeEntityModel;
 import net.minecraft.client.util.math.MatrixStack;
 
-public class TinyPotatoEntityModel extends CompositeEntityModel<TinyPotatoEntity> {
+import java.util.List;
+import java.util.Map;
 
-	private final ModelPart model;
+import static me.hydos.lint.util.ModelUtils.cuboid;
```
Comment (hYdos) [inline src/main/java/me/hydos/lint/client/entity/model/TinyPotatoEntityModel.java:32]: conc

File: src/main/java/me/hydos/lint/client/entity/model/KingTaterEntityModel.java [hunk 2]
```diff
@@ -21,106 +21,103 @@
...
-		left.addCuboid("left", -12.0F, -35.0F, 6.5F, 0, 1, 1, 0.0F, 31, 68);
-		left.addCuboid("left", -12.0F, -35.0F, 8.5F, 0, 1, 1, 0.0F, 31, 68);
-		left.addCuboid("left", -12.0F, -35.0F, 10.5F, 0, 1, 1, 0.0F, 31, 68);
+public class KingTaterEntityModel extends CompositeEntityModel<KingTaterEntity> {
+	private static final ModelPart root;
+	private static final int textureWidth = 96;
+	private static final int textureHeight = 82;
+	
+	static {
+		root = new ModelPart(List.of(), Map.of(
```
Comment (hYdos) [inline src/main/java/me/hydos/lint/client/entity/model/KingTaterEntityModel.java:41]: what the fuck is this formatting

File: src/main/java/me/hydos/lint/client/entity/model/BeeTaterEntityModel.java [hunk 3]
```diff
@@ -20,55 +20,68 @@
...
+	private static final ModelPart body;
+	private static final ModelPart left_wing;
+	private static final ModelPart right_wing;
+	private static final ModelPart leg1;
+	private static final ModelPart leg2;
+	private static final ModelPart leg3;
+	private static final int textureWidth = 64;
+	private static final int textureHeight = 64;
+	
+	// TODO: use ModelUtils.part to set pivot
```
Comment (hYdos) [inline src/main/java/me/hydos/lint/client/entity/model/BeeTaterEntityModel.java:44]: do this pls

**Comment by sylv256** [comment]
not finished yet crab

File: src/main/java/me/hydos/lint/bossbar/BossBarClientRenderer.java [hunk 4]
```diff
@@ -19,34 +19,25 @@
...
-			fill(stack, -140 + shadowOffsetX, 30 + shadowOffsetY, 140 + shadowOffsetX, 34 + shadowOffsetY, 0x80000000);
-		}
-		fill(stack, -140, 30, 140, 34, 0x991E1E1E);
-		fill(stack, -140, 30, bossBar.endX, 34, 0xFFE52320);
-		stack.pop();
-	}
-
-}
+//@Environment(EnvType.CLIENT)
+//public class BossBarClientRenderer extends DrawableHelper {
```
Comment (hYdos) [inline src/main/java/me/hydos/lint/bossbar/BossBarClientRenderer.java:23]: irr

**Review by hYdos — CHANGES_REQUESTED** [review]
a

File: src/main/java/me/hydos/lint/client/entity/model/KingTaterEntityModel.java (hunk 2 above)
Comment (sylv256) [reply src/main/java/me/hydos/lint/client/entity/model/KingTaterEntityModel.java:41]: best i can do until we do entitygenjson + blockbench plugin

Comment (hYdos) [reply src/main/java/me/hydos/lint/client/entity/model/KingTaterEntityModel.java:41]: wrong. do it the normal vanilla way

Comment (sylv256) [reply src/main/java/me/hydos/lint/client/entity/model/KingTaterEntityModel.java:41]: > wrong. do it the normal vanilla way

this is the vanilla way right?

File: src/main/resources/lint.aw [hunk 5]
```diff
@@ -6,10 +6,18 @@ accessible	method	net/minecraft/screen/ScreenHandler	addSlot	(Lnet/minecraft/scr
 accessible	method	net/minecraft/recipe/BrewingRecipeRegistry	registerPotionRecipe	(Lnet/minecraft/potion/Potion;Lnet/minecraft/item/Item;Lnet/minecraft/potion/Potion;)V
 accessible	method	net/minecraft/world/gen/feature/TreeFeature	isVine	(Lnet/minecraft/world/TestableWorld;Lnet/minecraft/util/math/BlockPos;)Z
 accessible	class	net/minecraft/recipe/CookingRecipeSerializer$RecipeFactory
-mutable field net/minecraft/client/sound/AbstractSoundInstance category Lnet/minecraft/sound/SoundCategory;
-accessible field net/minecraft/client/render/WorldRenderer playingSongs Ljava/util/Map;
-accessible field net/minecraft/world/gen/decorator/DecoratorContext generator Lnet/minecraft/world/gen/chunk/ChunkGenerator;
+mutable	field	net/minecraft/client/sound/AbstractSoundInstance category Lnet/minecraft/sound/SoundCategory;
```
Comment (hYdos) [inline src/main/resources/lint.aw:9]: based

**Review by hYdos — APPROVED** [review]

File: src/main/resources/lint.aw (hunk 5 above)
Comment (sylv256) [reply src/main/resources/lint.aw:9]: indeed

File: src/main/java/me/hydos/lint/block/organic/CorruptFlower.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		return LintBlocks.CORRUPT_GRASS.getDefaultState();
+	}
+
+	@Override
+	public void onEntityCollision(BlockState state, World world, BlockPos pos, Entity entity) {
+		if (entity instanceof LivingEntity livingEntity) {
+			if (((entity instanceof PlayerEntity) && ((PlayerEntity) entity).isCreative()) || (entity instanceof HostileEntity))
+				return;
+
+			entity.damage(DamageSource.CACTUS, 1.0F); // TODO: use something like SWEET_BERRY_BUSH instead of CACTUS
```
Comment (sylv256) [inline src/main/java/me/hydos/lint/block/organic/CorruptFlower.java]: @hYdos do you approve of this todo

Comment (hYdos) [reply src/main/java/me/hydos/lint/block/organic/CorruptFlower.java]: yes

Comment (sylv256) [reply src/main/java/me/hydos/lint/block/organic/CorruptFlower.java]: done

File: src/main/java/me/hydos/lint/entity/aggressive/I509VCBEntity.java [hunk 7]
```diff
@@ -28,20 +28,25 @@
...
-// TODO: Why does the boss stop levitating?
+// FIXME: Why does the boss stop levitating?
 public class I509VCBEntity extends HostileEntity {
+	private final ServerBossBar bossBar;
+
 	public I509VCBEntity(EntityType<? extends HostileEntity> entityType, World world) {
 		super(entityType, world);
-		ModernBossBar.getInstance().setTitle(getDisplayName().shallowCopy().styled(style -> style.withBold(true).withColor(TextColor.fromRgb(0xB41917))));
+//		ModernBossBar.getInstance().setTitle(getDisplayName().shallowCopy().styled(style -> style.withBold(true).withColor(TextColor.fromRgb(0xB41917))));
+		// FIXME: make BossBar color 0xB41917
```
Comment (sylv256) [inline src/main/java/me/hydos/lint/entity/aggressive/I509VCBEntity.java:48]: fix this soon #83

**Pull request merged**